### PR TITLE
feat: Support for the initial snippetgen phase 2 example

### PR DESCRIPTION
**Pull request by dazuma**
This PR implements enough of snippetgen phase 2 to generate the initial speech v1 test snippet for create_custom_class.

The first 17 commits are part of #890, #891, and #892. If those PRs are not yet merged, reviewers can ignore those commits and review only the final 5 commits that are unique to this PR.

The commits that matter are as follows. It may be easiest to review each commit independently.

1. Updated the googleapis/googleapis submodule and regenerated all goldens based on the update. This was needed because the current goldens referenced an older speech v1 that didn't include the adaptation service.
2. Fixed the file formatter so it actually respects the intended configs, and so that it performs all "safe" autocorrect operations. This is needed because we needed it to postprocess the generated snippet to fix a few rubocop style issues that are difficult to handle without having an actual Ruby parser. As part of this, I needed to make some adjustments to a few of the templates to fix existing layout/spacing issues without using rubocop because we're disabling the autocorrect for some files (mostly generated tests). This commit also regenerates the goldens again.
3. Updated the generator so it generates snippets based on any loaded configs. This effectively "turns on" phase 2.
4. Implemented a few more features that are used by the create_custom_class test. In particular, list and map iteration are now implemented. (And streaming request initialization is also implemented, since it's closely related and it's a good test case for iteration.) This commit also splits `Statement.Declaration` out of the statement presenter into its own presenter, because it's reused "raw" by iteration.
5. Regenerated the goldens once again, to generate the create_custom_class snippet.

### CODE REVIEW

**Review by amanda-tarafa — APPROVED** [review]
LGTM from the SnippetGen side of things.
This all looks great!

File: gapic-generator/templates/default/helpers/default_helper.rb [hunk 1]
```diff
@@ -66,4 +66,17 @@ def format_number value
   def assert_locals *locals
     locals.each { |local| raise "missing local in template" if local.nil? }
   end
+
+  def start_line_spacer name = :default
+    (@line_spacers ||= {})[name] = false
+  end
+
+  def line_spacer name = :default
```
Comment (viacheslav-rostovtsev) [inline gapic-generator/templates/default/helpers/default_helper.rb:74]: this is way too opaque. Do we really need it as a helper? If so, needs documenting

File: gapic-generator/lib/gapic/presenters/snippet/expression_presenter.rb [hunk 2]
```diff
@@ -32,7 +32,7 @@ class ExpressionPresenter
         # @param json [String]
         #     The JSON representation of the expression
         #
-        def initialize proto, json
+        def initialize proto, json # rubocop:disable Metrics/CyclomaticComplexity,Metrics/PerceivedComplexity
```
Comment (viacheslav-rostovtsev) [inline gapic-generator/lib/gapic/presenters/snippet/expression_presenter.rb:35]: Is this until the end of file?

**Review by viacheslav-rostovtsev — APPROVED** [review]
LGTM

**Pull request merged**